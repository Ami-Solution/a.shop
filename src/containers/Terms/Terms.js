import React from 'react';
import { withRouter } from 'react-router-dom';
import PropTypes from 'prop-types';

import tr from '../../../translate';
import Layout from '../../../components/Layout';
import { Padding } from '../../../components/Spaces';
import Mention from '../../../components/Mention';
import RoundIconBtn from '../../../components/RoundIconBtn';

const Terms = ({ history }) => (
  <Layout textAlignLeft isFullScreen>
    <Layout.Header>
      <Padding horizontal="m" top="m" bottom="xs">
        <RoundIconBtn onClick={history.goBack} type="prev" />
      </Padding>
    </Layout.Header>
    <Layout.ScrollableBody>
      <Padding horizontal="m" top="xs" bottom="m">
        <Mention>{tr('onboarding.terms')}</Mention>
      </Padding>
    </Layout.ScrollableBody>
  </Layout>
);

Terms.propTypes = {
  history: PropTypes.shape({
    goBack: PropTypes.func.isRequired
  }).isRequired,
};

export default withRouter(Terms);
