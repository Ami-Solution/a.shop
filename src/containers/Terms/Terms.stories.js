/* eslint-disable */
import React from 'react'
import styled, { css } from 'styled-components'
import { storiesOf } from '@storybook/react'

import Terms from './index'

storiesOf('04 - Screens/General', module).add('Terms', () => <Terms />)
