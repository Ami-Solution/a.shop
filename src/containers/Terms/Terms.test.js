/* global describe it expect */
import React from 'react';
import renderer from 'react-test-renderer';
import { MemoryRouter } from 'react-router'
import 'jest-styled-components';

import Terms from './index';

describe('Component Terms', () => {
  it('should render Terms', () => {
    const component = renderer.create(<MemoryRouter initialEntries={['/term']}><Terms /></MemoryRouter>);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
