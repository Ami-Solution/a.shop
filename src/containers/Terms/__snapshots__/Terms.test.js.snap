// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component Terms should render Terms 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-flow: column nowrap;
  -ms-flex-flow: column nowrap;
  flex-flow: column nowrap;
  -webkit-align-self: stretch;
  -ms-flex-item-align: stretch;
  align-self: stretch;
  min-height: 100%;
  width: 100%;
  text-align: center;
  position: relative;
  text-align: left;
  overflow: hidden;
  height: 100%;
  max-height: 100%;
}

.c1 {
  display: block;
}

.c7 {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  position: relative;
  overflow-x: hidden;
  overflow-y: auto;
}

.c2 {
  padding-left: 1.6rem;
  padding-right: 1.6rem;
  padding-top: 1.6rem;
  padding-bottom: 0.8rem;
}

.c8 {
  padding-left: 1.6rem;
  padding-right: 1.6rem;
  padding-top: 0.8rem;
  padding-bottom: 1.6rem;
}

.c9 {
  font-size: 1.2rem;
  color: #9B9B9B;
}

.c4 {
  display: block;
  overflow: visible;
  width: 1rem;
  height: 1rem;
}

.c5 {
  stroke: #9B9B9B;
  stroke-width: 0.2rem;
  fill: none;
}

.c6 {
  stroke: #9B9B9B;
  stroke-width: 0.2rem;
  fill: none;
}

.c3 {
  width: 2.4rem;
  height: 2.4rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  overflow: hidden;
  border-radius: 50%;
  border: solid 0.1rem #9B9B9B;
}

<div
  className="c0"
>
  <header
    className="c1"
  >
    <div
      className="c2"
    >
      <button
        className="c3"
        onClick={[Function]}
      >
        <svg
          className="c4"
          size="1rem"
          viewBox="0 0 12 12"
        >
          <line
            className="c5"
            vectorEffect="non-scaling-stroke"
            x1="0"
            x2="12"
            y1="6"
            y2="6"
          />
          <polyline
            className="c6"
            points="6 0 0 6 6 12"
            vectorEffect="non-scaling-stroke"
          />
        </svg>
      </button>
    </div>
  </header>
  <div
    className="c7"
  >
    <div
      className="c8"
    >
      <div
        className="c9"
      >
        {{ onboarding.terms }}
      </div>
    </div>
  </div>
</div>
`;
